src/libertem/io/dataset/mib.py

```diff
@@ -70,13 +80,39 @@ def is_valid_hdr(path):
         return line.startswith("HDR")
 
 
-def scan_size_from_hdr(hdr):
+def nav_shape_from_hdr(hdr):
     num_frames, scan_x = (
         int(hdr['Frames in Acquisition (Number)']),
         int(hdr['Frames per Trigger (Number)'])
     )
-    scan_size = (num_frames // scan_x, scan_x)
-    return scan_size
+    nav_shape = (num_frames // scan_x, scan_x)
+    return nav_shape
+
+
+def get_image_count_and_sig_shape(path):
+    path, ext = os.path.splitext(path)
+    ext = ext.lower()
+    if ext == '.mib':
+        pattern = "%s*.mib" % (
+            re.sub(r'[0-9]+$', '', path)
+        )
+    elif ext == '.hdr':
+        pattern = "%s*.mib" % path
+    else:
+        raise DataSetException("unknown extension")
```

Comment:
There is again some duplication with `_filenames`, could possibly be extracted as a helper function and called here instead. May also need to pass the (currently undocumented) `_disable_glob` parameter (use case: you have multiple `.mib` files from different data sets in the same directory, named with a number at the end, for example `dataset-1.mib`, `dataset-2.mib` etc. - this was solving a real, if uncommon, issue).